src/checkers/inference/util/JaifBuilder.java

```diff
@@ -98,9 +103,48 @@ public String createJaif() {
      */
     private void writeAnnotationHeader() {
         for (Class<? extends Annotation> annotation : supportedAnnotations) {
-            builder.append(buildAnnotationHeader(annotation));
-            builder.append("\n");
+            writeAnnotationHeader(annotation);
```

Comment:
java permits circularly dependent annotations, ie

```
@interface OneAnnot {
    TwoAnnot foo();
}

@interface TwoAnnot {
    OneAnnot bar();
}
```

However, CF issues warnings for circular dependencies [example](http://eisop.uwaterloo.ca/live/#mode=display&code=import+org.checkerframework.checker.nullness.qual.Nullable%3B%0Aclass+YourClassNameHere+%7B%0A++void+foo(Object+nn,+%40Nullable+Object+nbl)+%7B%0A++++nn.toString()%3B+//+OK%0A++++nbl.toString()%3B+//+Error%0A++%7D%0A%7D%0A%0A%40interface+InnerAnnot%7B%0A++++OtherAnnot+foo()%3B%0A%7D%0A%0A%40interface+OtherAnnot%7B%0A++++InnerAnnot+bar()%3B%0A%7D&typeSystem=nullness)

Do we need to update this algorithm to assume the possibility of circularly dependent annotations?

Reply:
Update: cyclicly dependent annotations are not allowed, but javac issues a weird error message, and also continues to annotation processing stage. This is a javac bug.